content/docs/user-guide/structure-of-remote-cache-by-example.md

```diff
@@ -0,0 +1,251 @@
+> - that you have set up an S3 bucket to be the remote cache
+> - you have downloaded an AWSAccessKeyId and a AWSSecretKey which you have configured on your local PC with `aws configure`.
+> - we will use a OS X/Linux CLI 
+
+
+## Set up a simple project
+
+To get started, we will set up a basic project tracked by both `git` and `dvc`. 
+The project will contain only data files and no pipelines, code or other anything else;
+the aim is to keep the project as simple as possible 
+so that the remote cache ideas can be clearly illustrated.
+
+Let's make a project called `project` and initialise it:
+
+```dvc
+$ mkdir project
+$ cd project
+$ git init
+$ dvc init
+```
+
+So far, I have not connected `git` to any external repository. 
+You can do that easily enough later on, but we won't worry about that here.
+
+What we do need is some data to add to the project. 
+In machine learning pipelines, data files can get quite large (e.g. hundreds of GB in size),
+but for our purposes they will be small text files so that commands run fast,
+and they will all contain the same content initially.
+
+Using the `echo` command to pipe text into some text files is one way to achieve this:
+
+```dvc
+$ echo data > data1
+$ echo data > data2
+$ echo data > data3
+$ echo data > data4
+```
+
+Now let's tell DVC about these new data files and ask for them to be specifically tracked:
+
+```dvc
+$ dvc add data1
+$ dvc add data2
+$ dvc add data3
+$ dvc add data4
+```
+
+Once that is done, you will notice that `.dvc` file is created for each of the original data files:
+
+```dvc
+$ ls
+data1 data2 data3 data4
+data1.dvc data2.dvc data3.dvc data4.dvc
+```
+
+The `dvc add` command should also have put all data files inside `.gitignore`, which is a good idea because if these four identical data files were in reality very big,
+it could take a long time to upload them to github and we want to avoid that.
+
+Let's check the `.gitignore` file then:
+
+```dvc
+% cat .gitignore
+/data1
+/data2
+/data3
+/data4
+```
+
+Thanksfully, the data files are all there!
+
+Now we can do an initial commit of everything else using `git`,
+but first, let's set up the connection between `dvc` and our remote S3 cache first. 
+
+When the remote cache is added, the file `.dvc/cache` will be updated with the remote cache details, so we will need to add these to git as well:
+
+```dvc
+$ dvc remote add s3cache s3://path/to/your/cache/
+$ git add .dvc/config
+```
+
+And finally let's finish with a initial commit:
+
+```dvc
+$ git commit -m "initial"
+```
+
+If you managed to get this far, we can now start talking about how the remote cache is structured in relation to our project files.
+
+## Pushing the DVC-tracked files to the remote cache
+
+Now come's the interesting part!
+
+> To fully appreciate how remote caches work, it's useful
+> to understand the concept of _indirection_. Indirection refers
+> to the naming of things using a reference.  
```

Comment:
Interesting 👍🏼 

A [link here](https://en.wikipedia.org/wiki/Indirection) would be useful too.

Reply:
Added the link and a better explanation.